src/core/payment/PaymentSummarizer.ts

```diff
@@ -0,0 +1,68 @@
+import { injectable } from "inversify";
+import { Block } from "../dataProvider/dataprovider-types";
+import { PayoutTransaction } from "../payoutCalculator/Model";
+import { ISummarizer, PaymentProcess } from "./Model";
+
+@injectable()
+export class PaymentSummarizer implements ISummarizer<PaymentProcess> {
+    async calculateTotals(base: PaymentProcess) : Promise<void> {
+        
+        const getTotalsFromBlocks = (blocks: Block[]) => {
+            let coinbasesum = 0
+            let feetransferfromcoinbasesum = 0
+            let usercommandtransactionfeessum = 0
+            blocks.forEach (block => {
+                coinbasesum += block.coinbase ?? 0
+                feetransferfromcoinbasesum += block.feetransferfromcoinbase ?? 0
+                usercommandtransactionfeessum += block.usercommandtransactionfees
+            })
+
+            return {coinbasesum, feetransferfromcoinbasesum, usercommandtransactionfeessum}
+        }
+
+        const getTotalPayouts = (transactions: PayoutTransaction[]) => {
+            let amountsum = 0
+            let feesum = 0
+
+            transactions.forEach(transaction => {
+                amountsum += transaction.amount ?? 0
+                feesum += transaction.fee ?? 0
+            })
+
+            return {amountsum,feesum}
+        }
+
+        const {coinbasesum, feetransferfromcoinbasesum, usercommandtransactionfeessum} = getTotalsFromBlocks(base.blocks)
+
+        const netcoinbasereceived = coinbasesum - feetransferfromcoinbasesum - usercommandtransactionfeessum
+
+        const {amountsum,feesum} = getTotalPayouts(base.payouts)
+
+        base.totals = {
+            coinBaseSum: coinbasesum,
+            freeTransferFromCoinBaseSum: feetransferfromcoinbasesum,
+            userCommandTransactionFeeSum: usercommandtransactionfeessum,
+            netCoinBaseReceveid: coinbasesum - feetransferfromcoinbasesum - usercommandtransactionfeessum,
```

Comment:
plus user command transaction fees (not minus.) The block producer gets to keep the transaction fees from the user transactions.

Reply:
I also don't see snark fees being accounted for.